File: multi/domain/src/main/java/com/example/domain/common/BaseEntity.java
package com.example.domain.common;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Getter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public class BaseEntity {

        @CreatedBy
        @Column(name = "created_by", updatable = false, length = 50)
        private String createdBy;

        @CreatedDate
        @Column(name = "created_at", nullable = false, updatable = false)
        private LocalDateTime createdAt;

        @LastModifiedBy
        @Column(name = "updated_by", length = 50)
        private String updatedBy;

        @LastModifiedDate
        @Column(name = "updated_at")
        private LocalDateTime updatedAt;


}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/common/BaseEntity.kt
package com.example.movie.persistence.common

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedBy
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@EntityListeners(AuditingEntityListener::class)
@MappedSuperclass
abstract class BaseEntity {
    @CreatedBy
    @Column(updatable = false)
    lateinit var createdBy: String

    @CreatedDate
    @Column(updatable = false)
    lateinit var createdAt: LocalDateTime

    @LastModifiedBy
    lateinit var updatedBy: String

    @LastModifiedDate
    lateinit var updatedAt: LocalDateTime

    fun update(updatedBy: String, updateAt: LocalDateTime) {
        this.updatedBy = updatedBy
        this.updatedAt = updatedAt
    }
}